app/services/agreements/base_seeder.rb

```diff
@@ -0,0 +1,59 @@
+module Agreements
+  class BaseSeeder
+    def initialize(rails_env: Rails.env, yaml_path: 'config')
+      @rails_env = rails_env
+      @yaml_path = yaml_path
+    end
+
+    def run
+      records.each do |key, config|
+        config = process_config(key, config)
+        record = record_class.find_or_initialize_by(
+          primary_attribute_bundle(config)
+        )
+        record.assign_attributes(config)
+        record.save!
+      end
+      after_seed
```

Comment:
I may want to add a `rescue` here to try and get better error messages (since right now a failed `find_by!` or `save!` doesn't always indicate where it failed).

Reply:
Apparently the `find_by!` exceptions don't give you any useful info so I'm going to skip this for now - we should really never get to this point anyway.